DatadogRUM/Sources/Integrations/TelemetryInterceptor.swift

```diff
@@ -20,6 +20,10 @@ internal struct TelemetryInterceptor: FeatureMessageReceiver {
         switch telemetry {
         case .error(let id, let message, let kind, let stack):
             interceptError(id: id, message: message, kind: kind, stack: stack)
+        case .metric(let metric) where metric.name == SDKMetricFields.UploadQuality.name:
+            interceptUploadQualityMetric(attributes: metric.attributes)
+            return true
+
```

Comment:
question/ The return true statement consumes the metric, meaning it will never be sent as a standalone metric event. As a result, the only effective operation here is intercepting metric.attributes.

Is this an intentional design choice? If so, could we document it more clearly? 🙏

For consistency, we might want to follow the convention of defining a namespace for each metric. Something like this could help clarify its purpose:
```swift
/// ...
/// This metric is not uploaded by the SDK. Instead, it is consumed internally  
/// within the message bus.
/// ...
enum UploadQualityMetric {
   /// Metric name
   public static let name = "upload_quality"

   // ...
}
```

Reply:
Yes, it is by design and lack documentation. It makes sense to have the `UploadQualityMetric` definition, I will add that 👍